Makefile

```diff
@@ -3,9 +3,12 @@ DEB_NAME := docker-push-latest-if-changed_$(VERSION)_all.deb
 
 .PHONY: builddeb
 builddeb:
-	mkdir -p dist
-	debuild -us -uc -b
-	mv ../$(DEB_NAME) dist/
+	docker build -f builddeb.Dockerfile -t builddeb .
+	mkdir -p dist/
+	docker run \
+		-v $(CURDIR):/mnt:rw \
```

Comment:
don't read/write mount the code -- make two separate mounts, one for the code, one for `dist` (having only `dist` be writeable)

this'll avoid a bunch of files owned by root in your workdir

Reply:
I did this since it needed to write stuff into the debian folder for whatever reason
```
(venv) andytran@dev13-uswest1cdevc:~/pg/docker_push_latest_if_changed/docker-push-latest-if-changed (build-deb-in-docker-container) $ make itest_trusty
docker build -f builddeb.Dockerfile -t builddeb .
Sending build context to Docker daemon  77.87MB
Step 1/3 : FROM        ubuntu:bionic
 ---> cd6d8154f1e1
Step 2/3 : RUN         apt-get update && apt-get install -y --no-install-recommends                 build-essential                 debhelper                 devscripts                 fakeroot             && apt-get clean
 ---> Using cache
 ---> 578bd7eae40f
Step 3/3 : WORKDIR     /mnt
 ---> Using cache
 ---> 04e6ddb5ff11
Successfully built 04e6ddb5ff11
Successfully tagged builddeb:latest
mkdir -p dist/
docker run \
                -v /nail/home/andytran/pg/docker_push_latest_if_changed/docker-push-latest-if-changed:/mnt:ro \
                -v /nail/home/andytran/pg/docker_push_latest_if_changed/docker-push-latest-if-changed/dist:/mnt/dist:rw \
                builddeb \
        /bin/bash -c "debuild -us -uc -b && mv ../docker-push-latest-if-changed_0.0.0_all.deb /mnt/dist"
 dpkg-buildpackage -rfakeroot -us -uc -ui -b
dpkg-buildpackage: warning: using a gain-root-command while being root
dpkg-buildpackage: info: source package docker-push-latest-if-changed
dpkg-buildpackage: info: source version 0.0.0
dpkg-buildpackage: info: source distribution xenial
dpkg-buildpackage: info: source changed by Chris Kuehl <ckuehl@yelp.com>
 dpkg-source --before-build mnt
dpkg-buildpackage: info: host architecture amd64
 fakeroot debian/rules clean
dh clean
   dh_auto_clean
   dh_clean
rm: cannot remove 'debian/.debhelper/generated/docker-push-latest-if-changed/installed-by-dh_installdocs': Read-only file system
dh_clean: rm -rf debian/.debhelper/ returned exit code 1
debian/rules:3: recipe for target 'clean' failed
make: *** [clean] Error 2
dpkg-buildpackage: error: fakeroot debian/rules clean subprocess returned exit status 2
debuild: fatal error at line 1152:
dpkg-buildpackage -rfakeroot -us -uc -ui -b failed
make: *** [builddeb] Error 29
```

I assumed it cleaned up everything after, but it still leaves some trash owned as root in the debian folder.
Going to just copy all the code in the Dockerfile and not deal with it while mounting the /dist folder as suggested.